src/geometry/manifold/instances/real.lean

```diff
@@ -93,8 +93,10 @@ def model_with_corners_euclidean_half_space (n : ℕ) [has_zero (fin n)] :
   end,
   right_inv'  := λx hx, update_eq_iff.2 ⟨max_eq_left hx, λ i _, rfl⟩,
   source_eq    := rfl,
-  unique_diff' := by simpa only [singleton_pi]
-    using unique_diff_on.pi (fin n) (λ _, ℝ) _ _ (λ i ∈ ({0} : set (fin n)), unique_diff_on_Ici 0),
+  unique_diff' :=
+    have this : unique_diff_on ℝ _ :=
+      unique_diff_on.pi (fin n) (λ _, ℝ) _ _ (λ i ∈ ({0} : set (fin n)), unique_diff_on_Ici 0),
+    by simpa only [singleton_pi] using this,
```

Comment:
I'm a bit worried having to change this means there are incompatible `module` instances now. On the other hand, there is only one distinct breakage, so the situation cannot be that bad.

Were these lines just relatively fragile already?

Reply:
The instances aren't really incompatible; they're still defeq, the unifier just has very slightly more trouble. Inferring instances around constant pi types like `(λ _, ℝ)` s always trouble, and the annotation `unique_diff_on ℝ _` this change required was a very simple one.